cli/deployment/deployment.go

```diff
@@ -0,0 +1,78 @@
+package deployment
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"strings"
+)
+
+// Deployment object represent real GCP Deployment entity that already created or have to be created
+// configFile field point to yaml file, generated from Config object with all cross-deployment references
+// overwritten with actual values
+type Deployment struct {
+	Outputs    map[string]string
+	config     Config
+	configFile string
+}
+
+// NewDeployment creates Deployment object and override all out refs, this means all
+// deployments it depends to should exists in GCP project
+func NewDeployment(config Config, outputs map[string]map[string]string) *Deployment {
+	file, err := ioutil.TempFile("", config.Name)
+	defer file.Close()
+	if err != nil {
+		log.Fatalf("Error creating temp file for deployment: %s, error: %v", config.FullName(), err)
+	}
+
+	data := replaceOutRefs(config, outputs)
+
+	_, err = file.Write(data)
+	if err != nil {
+		log.Fatalf("Error wirte to file: %s, error: %v", file.Name(), err)
+	}
+
+	return &Deployment{
+		config:     config,
+		configFile: file.Name(),
+	}
+}
+
+// String representation of Deployment instance
+func (d Deployment) String() string {
+	return fmt.Sprintf("Deployment[name=%s, config=%s]", d.config.FullName(), d.configFile)
+}
+
+// same as deployment.Config.FullName(), can be used in map[string]Deployment as a key
```

Comment:
This should be a full sentence, and should duplicate the text for `config.FullName`. The generated documentation won't be cross-referenced, so it would be helpful for someone looking here to see the full behavior description rather than needing to go find the `FullName` method on the `config` type.